sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSenderAsyncClient.java

```diff
@@ -342,7 +375,83 @@ public String getEntityPath() {
 
         return connectionProcessor
             .flatMap(connection -> connection.getManagementNode(entityName, entityType))
-            .flatMap(managementNode -> managementNode.cancelScheduledMessage(sequenceNumber, linkName.get()));
+            .flatMap(managementNode -> managementNode.cancelScheduledMessage(sequenceNumber, linkName.get(),
+                null));
+    }
+
+    /**
+     * Cancels the enqueuing of an already scheduled message, if it was not already enqueued.
+     *
+     * @param sequenceNumber of the scheduled message to cancel.
+     * @param transactionContext to be set on batch message before scheduling them on Service Bus.
+     *
+     * @return The {@link Mono} that finishes this operation on service bus resource.
+     *
+     * @throws IllegalArgumentException if {@code sequenceNumber} is negative.
+     */
+    public Mono<Void> cancelScheduledMessage(long sequenceNumber, ServiceBusTransactionContext transactionContext) {
```

Comment:
IIRC you said that your tests were failing for this? If that's the case, we should remove this.

Reply:
removing